backend/src/news/router.py

```diff
@@ -90,7 +90,27 @@ async def news_summary(
 ):
     response = {}
 
-    result = llm_client.generate_summary(payload.content)
+    result = openai_client.generate_summary(payload.content)
+    if result:
+        response["summary"] = result["影響"]
+        response["reason"] = result["原因"]
+    return response
+
+@router.post("/news_summary_custom_model")
+async def news_summary_with_custom_model(
+        payload: NewsSummaryCustomModelRequestSchema, user=Depends(authenticate_user_token)
+):
+    response = {}
+
+    if not payload.llm_model:
+        return {"message": "Model is required."}
+    elif payload.llm_model.lower() == "openai":
+        result = openai_client.generate_summary(payload.content)
+    elif payload.llm_model.lower() == "anthropic" or payload.llm_model.lower() == "claude":
+        result = anthropic_client.generate_summary(payload.content)
+    else:
+        return {"message": "Invalid model."}
+
```

Comment:
The business logic of the endpoints `/news_summary` and `/news_summary_custom_model` is very similar. Were I you, I would wrap them into a function which takes a `model` parameter to avoid repetition.